Microsoft.Toolkit.Uwp.UI.Animations/Implicit.cs

```diff
@@ -0,0 +1,249 @@
+// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using Windows.UI.Composition;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Hosting;
+
+namespace Microsoft.Toolkit.Uwp.UI.Animations
+{
+    /// <summary>
+    /// Attached Properties to enable Implicit Animations through XAML
+    /// </summary>
+    public class Implicit
+    {
+        /// <summary>
+        /// Identifies the Implicit.ShowAnimations XAML attached property
+        /// </summary>
+        public static readonly DependencyProperty ShowAnimationsProperty =
+            DependencyProperty.RegisterAttached("ShowAnimations", typeof(AnimationCollection), typeof(Implicit), new PropertyMetadata(null, ShowAnimationsChanged));
+
+        /// <summary>
+        /// Identifies the Implicit.HideAnimations XAML attached property
+        /// </summary>
+        public static readonly DependencyProperty HideAnimationsProperty =
+            DependencyProperty.RegisterAttached("HideAnimations", typeof(AnimationCollection), typeof(Implicit), new PropertyMetadata(null, HideAnimationsChanged));
+
+        /// <summary>
+        /// Identifies the Implicit.Animations XAML attached property
+        /// </summary>
+        public static readonly DependencyProperty AnimationsProperty =
+            DependencyProperty.RegisterAttached("Animations", typeof(AnimationCollection), typeof(Implicit), new PropertyMetadata(null, AnimationsChanged));
+
+        /// <summary>
+        /// Gets the value of the Implicit.ShowAnimations XAML attached property.
+        /// </summary>
+        /// <param name="obj">The <see cref="FrameworkElement"/> to get the value from</param>
+        /// <returns><see cref="AnimationCollection"/></returns>
+        public static AnimationCollection GetShowAnimations(DependencyObject obj)
+        {
+            var collection = (AnimationCollection)obj.GetValue(ShowAnimationsProperty);
+
+            if (collection == null)
```

Comment:
You can't prevent an attached property from being set to null even if you throw, because the throw would done after the set:
```
try {
    myObject.SetValue(Implicit.AnimationsProperty, null);
} catch { }
```